modules/memberships/class-jetpack-memberships.php

```diff
@@ -277,5 +288,55 @@ public static function get_blog_id() {
 	public static function get_connected_account_id() {
 		return get_option( self::$connected_account_id_option_name );
 	}
+
+	/**
+	 * Whether Recurring Payments are enabled.
+	 *
+	 * @return bool
+	 */
+	public static function is_enabled_jetpack_recurring_payments() {
+		// For WPCOM sites.
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_any_blog_stickers' ) ) {
+			$site_id = get_current_blog_id();
+			return has_any_blog_stickers( array( 'personal-plan', 'premium-plan', 'business-plan', 'ecommerce-plan' ), $site_id );
+		}
+
+		// For Jetpack sites.
+
+		// FIXME: We enable Recurring Payments on all Jetpack sites, including those on a Free plan.
+		// The reason is that in Jetpack, we cannot yet support the 'new', Upgrade and Stripe Nudge-based
+		// flow, since the post-checkout redirect skips the Jetpack Onboarding Checklist in Calypso,
+		// where Akismet and VaultPress installation is triggered.
+		// Instead, we continue to show the legacy, in-block prompts to upgrade the site's plan, and to
+		// connect Stripe. This however means that we have to display the block also if the site's on a Free plan.
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, uncomment the right-hand side of the line with the
+		// return statement below.
+		// phpcs:ignore
+		return Jetpack::is_active(); // && Jetpack_Plan::supports( 'recurring-payments' ); // phpcs:ignore
```

Comment:
Here we're hitting the situation where the upgrade nudge is not an enhancement, but an alternative. We should be inspecting the filter version to know whether we should mark unavailable —in which case the nudge will be applied— or circumvent the nudge mechanism —in which case the original behavior is maintained.

We'll probably need to pull the filter check into a dedicated method to keep it's default in a single place:

```php
public static function is_upgrade_nudge_available() {
  return apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false );
}
```

Then we can check it here:

```suggestion
		return Jetpack::is_active() && (
			! Jetpack_Gutenberg::is_upgrade_nudge_available() ||
			Jetpack_Plan::supports( 'recurring-payments' )
		);
```

Reply:
We can do that; not sure how to best handle the [client side](https://github.com/Automattic/jetpack/pull/13394#discussion_r321776190) though.

I was cutting some corners here, since doing this "right" seemed to require a rather complex solution for what I still hope to be somewhat temporary (hoping to remove all the `this.isJetpack &&` stuff once the backend installer is sorted out and we'll be able to enable nudges in Jetpack).

Reply:
Did you consider something like the suggestion above?

Reply:
Ahh, missed it, my bad. Makes a lot of sense, I'll apply it.